### PR TITLE
Hook clients separation

### PR DESCRIPTION
**Pull request by Udith-Gayan**

### CODE REVIEW

File: src/clients/hook-clients/governor-client.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const { DefaultValues } = require("../../defaults");
+
+const GovernorEvents = {
+    RegistryInitialized: EvernodeEvents.RegistryInitialized
+}
+
+class GovernorClient {
```
Comment (kithminisg) [inline src/clients/hook-clients/governor-client.js:7]: _**extends**_ keyword missing in the class declaration.

File: package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "version": "0.5.16",
     "scripts": {
         "lint": "./node_modules/.bin/eslint src/**/*.js",
-        "build": "npm run lint && ncc build src/index.js -e elliptic -e xrpl -e ripple-address-codec -e ripple-keypairs -o dist/",
+        "build": "npm run lint && ncc build -C src/index.js -e elliptic -e xrpl -e ripple-address-codec -e ripple-keypairs -o dist/",
```
Comment (kithminisg) [inline package.json:12]: Let's keep the previous build command byremoving `-C`

File: src/evernode-common.js [hunk 3]
```diff
@@ -68,9 +68,8 @@ const HookStateKeys = {
     MAX_TOLERABLE_DOWNTIME: "455652010000000000000000000000000000000000000000000000000000000A",
     MOMENT_TRANSIT_INFO: "455652010000000000000000000000000000000000000000000000000000000B",
     MAX_TRX_EMISSION_FEE: "455652010000000000000000000000000000000000000000000000000000000C",
-    REGISTRY_ADDR: "455652010000000000000000000000000000000000000000000000000000000D",
-    HEARTBEAT_HOOK_ADDR: "455652010000000000000000000000000000000000000000000000000000000E",
-
+    REGISTRY_ADDR: "455652010000000000000000000000000000000000000000000000000000000E",
```
Comment (kithminisg) [inline src/evernode-common.js:71]: Shall we change the order of new state keys inside  `HookStateKeys` ?  `45565201....D` followed by `45565201...E`

File: src/clients/hook-clients/hook-client-factory.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+const { HookAccountTypes } = require("../../defaults");
+const { RegistryClient } = require("./registry-client");
+const { GovernorClient } = require("./governor-client");
+const { HeartbeatClient } = require("./heartbeat-client");
+
+class HookClientFactory {
+    static async create(hookAccountType) {
+        let hookAccount;
```
Comment (chalith) [inline src/clients/hook-clients/hook-client-factory.js:8]: hookClient

File: src/clients/hook-clients/hook-client-factory.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+const { HookAccountTypes } = require("../../defaults");
+const { RegistryClient } = require("./registry-client");
+const { GovernorClient } = require("./governor-client");
+const { HeartbeatClient } = require("./heartbeat-client");
+
+class HookClientFactory {
+    static async create(hookAccountType) {
```
Comment (chalith) [inline src/clients/hook-clients/hook-client-factory.js:7]: Shall we change hookAccountType->hookType

File: src/clients/hook-clients/governor-client.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+const GovernorEvents = {
+    RegistryInitialized: EvernodeEvents.RegistryInitialized
+}
+
+class GovernorClient extends BaseEvernodeClient {
+    constructor(options = {}) {
+        super((options.governorAddress || DefaultValues.governorAddress), null, Object.values(GovernorEvents), false, options);
+    }
+
+    on(event, handler) {
```
Comment (chalith) [inline src/clients/hook-clients/governor-client.js:14]: Do we need to override this since it's in the base class as well

File: src/defaults.js [hunk 7]
```diff
@@ -1,10 +1,16 @@
 const DefaultValues = {
-    registryAddress: 'raaFre81618XegCrzTzVotAmarBcqNSAvK',
+    governorAddress: 'rDxkQ7Jaq1igBmNNavXqsZ5vyEoYRKgT8B',
     rippledServer: 'wss://hooks-testnet-v2.xrpl-labs.com',
     xrplApi: null,
     stateIndexId: 'evernodeindex'
 }
 
+const HookAccountTypes = {
```
Comment (chalith) [inline src/defaults.js:8]: What if we make this const like this
```
const HookTypes = {
    governor: 'GOVERNOR',
    registry: 'REGISTRY',
    heartbeat: 'HEARTBEAT'
}
```

File: src/state-helpers.js [hunk 8]
```diff
@@ -328,7 +329,7 @@ class StateHelpers {
             Buffer.from(HookStateKeys.MOMENT_TRANSIT_INFO, 'hex').compare(stateKey) === 0 ||
             Buffer.from(HookStateKeys.MAX_TRX_EMISSION_FEE, 'hex').compare(stateKey) === 0 ||
             Buffer.from(HookStateKeys.REGISTRY_ADDR, 'hex').compare(stateKey) === 0 ||
-            Buffer.from(HookStateKeys.HEARTBEAT_HOOK_ADDR, 'hex').compare(stateKey) === 0) {
+            Buffer.from(HookStateKeys.HEARTBEAT_ADDR, 'hex').compare(stateKey) === 0) {
```
Comment (chalith) [inline src/state-helpers.js:332]: There's registry address and heartbeat address twice

**Review by chalith — APPROVED** [review]

**Review by kithminisg — APPROVED** [review]

**Pull request merged**